### PR TITLE
Allow cookbook to install 1.0.0

### PR DESCRIPTION
**Pull request by bdangit**
* [major] Support for 1.0.x which is **not backward compatible**
* [major] Updated apt cookbook pin to ~> 4.0
* [major] Updated yum cookbook pin to ~> 4.0
* [minor] Reduce number of TK Cases to 12.1, 12.7 and 12.14.
* [patch] removed specific branch used for `compat_resource`

### CODE REVIEW

**Comment by bdangit** [comment]
Alright.  This is ready for review @fishnix and anyone else that just wants to eyeball the changes.

For whatever reason, my wercker hook is defunct.  Althought it automatically triggered a PR build: https://app.wercker.com/bdangit/chef-influxdb/runs/build/57e7758266c58c0100e24c64

**Comment by fishnix** [comment]
Cool @bdangit, I'll have a look at it in the next day or so!

File: resources/install.rb [hunk 1]
```diff
@@ -89,11 +89,11 @@
     end
   elsif node.platform_family? 'debian'
     apt_repository 'influxdb' do
-      action :delete
+      action :remove
       only_if { include_repository }
     end
   else
-    Chef.Log.warn "I do not support your platform: #{node.platform_family}"
+    Chef.Log.warn "I do not support your platform: #{node['platform_family']}"
```
Comment (fishnix) [inline resources/install.rb]: Should we be raising here on failure to remove?

**Comment by fishnix** [comment]
@bdangit this looks good to me, I ran through almost all of the kitchen scenarios 👍

**Comment by bdangit** [comment]
alright merging this all in.

thanks for the review, @fishnix @julian7

**Comment by bdangit** [comment]
fixes #125 #123 #114

**Pull request merged**